### PR TITLE
W8 - TungNguyen - Broadcast Receiver

### PR DESCRIPTION
**Pull request by at-tungnguyen2**
#### Related issues (Optional)
Task #

#### Description (Required)
- [x] Use broadcast receiver

### CODE REVIEW

File: app/src/at-tungnguyen/java/vn/asiantech/internship/ui/broadcast/BroadcastActivity.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+            case BatteryManager.BATTERY_HEALTH_UNSPECIFIED_FAILURE:
+                mBatteryHealth.setText(R.string.battery_heath_failure);
+                break;
+        }
+    }
+
+    /**
+     * Register battery
+     */
+    private void BatteryLevelReceiver() {
```
Comment (at-dev-support-01) [inline app/src/at-tungnguyen/java/vn/asiantech/internship/ui/broadcast/BroadcastActivity.java]: ERROR: [MethodNamingConventions] Method names should not start with capital letters
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#MethodNamingConventions

**Comment by at-tungnguyen2** [comment]
@at-quocnguyen Please review for me, thank you

File: app/src/at-tungnguyen/AndroidManifest.xml [hunk 2]
```diff
@@ -38,5 +38,7 @@
         <activity android:name=".ui.viewpager.ui.ViewPagerActivity" />
         <activity android:name=".ui.viewpager.ui.TabLayoutActivity" />
         <activity android:name=".ui.unittest.LoginActivity" />
+        <activity android:name=".ui.broadcast.BroadcastActivity"/>
     </application>
-</manifest>
+
+</manifest>
```
Comment (at-quocnguyen) [inline app/src/at-tungnguyen/AndroidManifest.xml]: break line

File: app/src/at-tungnguyen/java/vn/asiantech/internship/ui/viewpager/model/Broadcast.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+
+/**
+ * Author Asian Tech Inc.
+ * Created by tungnguyen on 10/01/2018.
+ */
+
+public class Broadcast extends BroadcastReceiver {
```
Comment (at-quocnguyen) [inline app/src/at-tungnguyen/java/vn/asiantech/internship/ui/viewpager/model/Broadcast.java]: This file used for what ??? If don't used please remove file

File: app/src/at-tungnguyen/res/drawable-xxhdpi/bg_battery_100.xml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<animation-list xmlns:android="http://schemas.android.com/apk/res/android"
+    android:id="@+id/selected"
+    android:oneshot="false">
+    <item
+        android:drawable="@drawable/ic_battery_full"
+        android:duration="500" />
+
+</animation-list>
```
Comment (at-quocnguyen) [inline app/src/at-tungnguyen/res/drawable-xxhdpi/bg_battery_100.xml]: break line

File: app/src/at-tungnguyen/res/drawable-xxhdpi/bg_battery_20.xml [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    <item
+        android:drawable="@drawable/ic_battery_charing_2"
+        android:duration="500" />
+    <item
+        android:drawable="@drawable/ic_battery_charing_3"
+        android:duration="500" />
+    <item
+        android:drawable="@drawable/ic_battery_full"
+        android:duration="500" />
+</animation-list>
```
Comment (at-quocnguyen) [inline app/src/at-tungnguyen/res/drawable-xxhdpi/bg_battery_20.xml]: break line

File: app/src/at-tungnguyen/res/drawable-xxhdpi/bg_battery_50.xml [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+    <item
+        android:drawable="@drawable/ic_battery_charing_2"
+        android:duration="500" />
+    <item
+        android:drawable="@drawable/ic_battery_charing_3"
+        android:duration="500" />
+    <item
+        android:drawable="@drawable/ic_battery_full"
+        android:duration="500" />
+</animation-list>
```
Comment (at-quocnguyen) [inline app/src/at-tungnguyen/res/drawable-xxhdpi/bg_battery_50.xml]: break line

File: app/src/at-tungnguyen/res/drawable-xxhdpi/bg_battery_80.xml [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+    android:id="@+id/selected"
+    android:oneshot="false">
+    <item
+        android:drawable="@drawable/ic_battery_charing_3"
+        android:duration="500" />
+    <item
+        android:drawable="@drawable/ic_battery_full"
+        android:duration="500" />
+
+</animation-list>
```
Comment (at-quocnguyen) [inline app/src/at-tungnguyen/res/drawable-xxhdpi/bg_battery_80.xml]: break line

File: app/src/at-tungnguyen/res/drawable-xxhdpi/spin_animation.xml [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+        android:drawable="@drawable/ic_battery_charing_2"
+        android:duration="200" />
+    <item
+        android:drawable="@drawable/ic_battery_charing_3"
+        android:duration="200" />
+    <item
+        android:drawable="@drawable/ic_battery_full"
+        android:duration="200" />
+
+</animation-list>
```
Comment (at-quocnguyen) [inline app/src/at-tungnguyen/res/drawable-xxhdpi/spin_animation.xml]: break line

File: app/src/at-tungnguyen/java/vn/asiantech/internship/ui/viewpager/ui/BroadCastFragment.java [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+           @Override
+           public void onClick(View view) {
+               Intent i = new Intent(getContext(), BroadcastActivity.class);
+               startActivity(i);
+           }
+       });
+        return view;
+    }
+
+
```
Comment (at-quocnguyen) [inline app/src/at-tungnguyen/java/vn/asiantech/internship/ui/viewpager/ui/BroadCastFragment.java]: remove 2 line

File: app/src/at-tungnguyen/java/vn/asiantech/internship/ui/broadcast/BroadcastActivity.java [hunk 10]
```diff
@@ -0,0 +1,167 @@
...
+                mImageView.setBackgroundResource(R.drawable.ic_battery_charing_1);
+            } else if (levelBattery > 20 && levelBattery < 50) {
+                mImageView.setBackgroundResource(R.drawable.ic_battery_charing_2);
+            } else if (levelBattery < 80) {
+                mImageView.setBackgroundResource(R.drawable.ic_battery_charing_3);
+            } else if (levelBattery == 100) {
+                mImageView.setBackgroundResource(R.drawable.ic_battery_full);
+            }
+        }
+
```
Comment (at-quocnguyen) [inline app/src/at-tungnguyen/java/vn/asiantech/internship/ui/broadcast/BroadcastActivity.java]: remove line

**Comment by at-tungnguyen2** [comment]
@at-quocnguyen  Please check for me, thank you

**Pull request merged**